lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -66,26 +64,28 @@ defmodule LambdaEthereumConsensus.ForkChoice do
   @spec init({BeaconState.t(), SignedBeaconBlock.t(), Types.uint64()}) ::
           {:ok, Store.t()} | {:stop, any}
   def init({anchor_state = %BeaconState{}, signed_anchor_block = %SignedBeaconBlock{}, time}) do
-    result =
-      case Helpers.get_forkchoice_store(anchor_state, signed_anchor_block.message) do
-        {:ok, store = %Store{}} ->
-          store = Handlers.on_tick(store, time)
-          Logger.info("[Fork choice] Initialized store.")
-          {:ok, store}
-
-        {:error, error} ->
-          {:stop, error}
-      end
-
-    # TODO: this should be done after validation
-    :ok = StateStore.store_state(anchor_state)
-    :ok = BlockStore.store_block(signed_anchor_block)
-
-    slot = signed_anchor_block.message.slot
-    :telemetry.execute([:sync, :store], %{slot: slot})
-    :telemetry.execute([:sync, :on_block], %{slot: slot})
-
-    result
+    case Helpers.get_forkchoice_store(anchor_state, signed_anchor_block.message) do
+      {:ok, %Store{} = store} ->
+        Logger.info("[Fork choice] Initialized store.")
+
+        slot = signed_anchor_block.message.slot
+        :telemetry.execute([:sync, :store], %{slot: slot})
+        :telemetry.execute([:sync, :on_block], %{slot: slot})
+
+        # TODO: solve this in a more elegant way
+        [anchor_block_root] = Map.keys(store.blocks)
+
+        store
+        |> Map.delete(:blocks)
```

Comment:
instead of deleting this explicitly, shouldn't `get_forkchoice_store` also call store getters/setters?

Reply:
because otherwise the abstraction is leaking here

Reply:
Done!